.changelog/2315.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+resource/mongodbatlas_network_peering: Fixes computed values of `altas_gcp_project_id` and `atlas_vpc_name` by obtaining values from associated container
+```
```

Comment:
we can consider this a bug fix as it reverts to the correct behaviour we had in 1.16.0, but open to other opinions.